libs/workspace/src/lint/mod.rs

```diff
@@ -195,9 +212,12 @@ impl<D> LintManager<D> {
                     }));
                 }
             }
+            if config.enabled() {
+                enabled_lints.push(lint);
```

Comment:
Disabled lints will now be filtered here instead of being checked multiple times below